tests/org.pitest.pitclipse.ui.tests/src/org/pitest/pitclipse/ui/behaviours/pageobjects/RunConfigurationSelector.java

```diff
@@ -59,25 +70,225 @@ public PitRunConfiguration getConfiguration(String configName) {
+        bot.radio(mutatorGroup.getDescriptor()).click();
+        closeConfigurationShell();
+    }
+
+    public void setAdditionalCustomMutator(String configurationName, Mutators mutator) {
+        activateMutatorsTab(configurationName);
+        SWTBotRadio radioButton = bot.radio(PitMutatorsTab.CUSTOM_MUTATOR_RADIO_TEXT);
+        if (!radioButton.isSelected()) {
+            radioButton.click();
+        }
+        SWTBotTable table = bot.table();
+        setMutator(table, mutator);
+        closeConfigurationShell();
+    }
+
+    public void setOneCustomMutator(String configurationName, Mutators mutator) {
+        activateMutatorsTab(configurationName);
+        bot.radio(PitMutatorsTab.CUSTOM_MUTATOR_RADIO_TEXT).click();
+        uncheckAllMutators();
+        SWTBotTable table = bot.table();
+        setMutator(table, mutator);
+        closeConfigurationShell();
+    }
+
+    public void checkAllMutators(String configurationName) {
+        activateMutatorsTab(configurationName);
+        bot.radio(PitMutatorsTab.CUSTOM_MUTATOR_RADIO_TEXT).click();
+        SWTBotTable table = bot.table();
+        Display.getDefault().syncExec(() -> {
+            for (TableItem item : table.widget.getItems()) {
+                item.setChecked(true);
+            }
+        });
+        closeConfigurationShell();
+    }
+
+    /**
+     * Assumes the Mutator tab is active and custom mutator mode is selected.
+     */
+    private void uncheckAllMutators() {
+        SWTBotTable table = bot.table();
+        Display.getDefault().syncExec(() -> {
+            for (TableItem item : table.widget.getItems()) {
+                item.setChecked(false);
+            }
+        });
+        // don't close, because you cannot apply empty mutators
+    }
+
+    /**
+     * Assumes the Mutator tab is active and custom mutator mode is selected.
+     * @param table   where to check the mutator
+     * @param mutator which is set to active
+     */
+    private void setMutator(SWTBotTable table, Mutators mutator) {
+        Display.getDefault().syncExec(() -> {
+            for (TableItem item : table.widget.getItems()) {
+                if (item.getData().equals(mutator.name())) {
+                    item.setChecked(true);
+                    break;
+                }
+            }
+        });
+    }
+
+    /**
+     * Closes the configuration shell and applies, if possible.
+     */
+    private void closeConfigurationShell() {
+        SWTBotShell shell = bot.shell(RUN_CONFIGURATIONS);
+        SWTBotButton apply = shell.bot().button("Apply");
+        if (apply.isEnabled()) {
+            apply.click();
+        }
+        shell.close();
+    }
+
+    public void runWithConfiguration(String configurationName) {
+        activateConfiguration(configurationName);
+        SWTBotShell shell = bot.shell(RUN_CONFIGURATIONS);
+        shell.bot()
+                .button(RUN)
+                .click();
+        shell.close();
```

Comment:
shouldn't clicking on button "Run" already close the dialog?
If that's the case I fear that also closing the shell might break something... maybe you should not close it like that, and instead wait for the dialog to be closed: `bot.waitUntil(Conditions.shellCloses(shell));`

Reply:
OK, let's hope not to have another synchronization issue here ;)
probably the dialog will close fast enough.
In general, in SWTBot, it's usually best practice to wait for a dialog to close, but I'm not sure we do that consistently in this code base